File: generate-assets-list.py
#!/usr/bin/env python3
"""
Build-time script to generate assets-list.json for GitHub Pages
This scans the assets folder and creates a JSON file with all PDF files
"""

import os
import json
from pathlib import Path

def generate_assets_list():
    print('🔍 Scanning assets folder for PDF files...')
    
    # Get the script directory
    script_dir = Path(__file__).parent
    assets_dir = script_dir / 'assets'
    output_file = script_dir / 'assets-list.json'
    
    try:
        # Check if assets directory exists
        if not assets_dir.exists():
            print('❌ Assets directory not found')
            return
        
        # Get all PDF files
        pdf_files = []
        for file_path in assets_dir.glob('*.pdf'):
            if file_path.is_file():
                pdf_files.append(file_path.name)
        
        # Sort files alphabetically
        pdf_files.sort()
        
        print(f'✅ Found {len(pdf_files)} PDF files:')
        for file in pdf_files:
            print(f'   📄 {file}')
        
        # Write JSON file
        with open(output_file, 'w', encoding='utf-8') as f:
            json.dump(pdf_files, f, indent=2)
        
        print(f'✅ Generated assets-list.json with {len(pdf_files)} files')
        print(f'📁 Output: {output_file}')
        
    except Exception as error:
        print(f'❌ Error generating assets list: {error}')
        exit(1)

if __name__ == '__main__':
    generate_assets_list() 

File: generate-assets-list.js
#!/usr/bin/env node
/**
 * Build-time script to generate assets-list.json for GitHub Pages
 * This scans the assets folder and creates a JSON file with all PDF files
 */

const fs = require('fs');
const path = require('path');

function generateAssetsList() {
  console.log('🔍 Scanning assets folder for PDF files...');
  
  const assetsDir = path.join(__dirname, 'assets');
  const outputFile = path.join(__dirname, 'assets-list.json');
  
  try {
    // Check if assets directory exists
    if (!fs.existsSync(assetsDir)) {
      console.log('❌ Assets directory not found');
      return;
    }
    
    // Read all files in assets directory
    const files = fs.readdirSync(assetsDir);
    
    // Filter for PDF files
    const pdfFiles = files.filter(file => {
      const ext = path.extname(file).toLowerCase();
      return ext === '.pdf';
    });
    
    // Sort files alphabetically
    pdfFiles.sort();
    
    console.log(`✅ Found ${pdfFiles.length} PDF files:`);
    pdfFiles.forEach(file => console.log(`   📄 ${file}`));
    
    // Write JSON file
    fs.writeFileSync(outputFile, JSON.stringify(pdfFiles, null, 2));
    
    console.log(`✅ Generated assets-list.json with ${pdfFiles.length} files`);
    console.log(`📁 Output: ${outputFile}`);
    
  } catch (error) {
    console.error('❌ Error generating assets list:', error.message);
    process.exit(1);
  }
}

// Run the script
generateAssetsList(); 